### ISSUE TITLE
Threaded comments on github issues and sort functionality

### ISSUE THREAD

**Issue by Drakedude** [isaacs/github#837]
Particularly detailed or controversial issues, such as those here, can accrue a lot of comments. This will make conversations much easier to follow and triangulate valuable contributions.

**Comment by cirosantilli** [comment]
Yeah, just copy reddit to start with.

**Comment by diegocam** [comment]
Yes please add threads to all comments (in Issues, PRs, etc..). Its very annoying to go through an issue's comment history where non related comments are scattered, or where "quote reply" comments have entire comments attached to them. 

Something like this would be great:

<img width="920" alt="Screen Shot 2021-05-29 at 8 07 10 PM" src="https://user-images.githubusercontent.com/4230798/120088895-e0a08b80-c0ba-11eb-9dcd-2e4cf7f6a92b.png">
